cvmfs/file_processing/chunk.cc

```diff
@@ -63,8 +63,14 @@ void Chunk::Initialize() {
   zlib_context_.opaque   = Z_NULL;
   zlib_context_.next_in  = Z_NULL;
   zlib_context_.avail_in = 0;
-  const int zlib_retval = deflateInit(&zlib_context_, Z_DEFAULT_COMPRESSION);
-  assert(zlib_retval == 0);
+  if (compression_alg_ == zlib::kZlibDefault) {
+    const int zlib_retval = deflateInit(&zlib_context_, Z_DEFAULT_COMPRESSION);
+    assert(zlib_retval == 0);
+  } else if (compression_alg_ == zlib::kNoCompression) {
+    // Initialize the zlib library to no compression
+    const int zlib_retval = deflateInit(&zlib_context_, 0);
```

Comment:
Yes, it does.  This was an incomplete patch set.  I'm working (and debugging) another patch set now that should make this much more modular.  I'm getting some sort of error though here: https://github.com/djw8605/cvmfs/blob/debug_compress/cvmfs/compression.cc#L816